### PR TITLE
Update @vector-im/compound-design-tokens

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

The changes:
- The heading sm was fixed in this design tokens PR: https://github.com/element-hq/compound-design-tokens/pull/96 
   The playwright snapshots are updated to use the correct heading.
- The icons are also updated https://github.com/element-hq/compound-design-tokens/pull/95
   The jest snapshots are updated for theses icons

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -6946,14 +6946,6 @@ matrix-widget-api@^1.8.2:
     "@types/events" "^3.0.0"
     events "^3.2.0"
 
-matrix-widget-api@^1.8.2:
```
Comment (florianduros) [inline yarn.lock:6949]: Duplicate. Already present line 6941.

File: playwright/snapshots/spaces/spaces.spec.ts/space-panel-expanded-linux.png
Comment (florianduros) [inline playwright/snapshots/spaces/spaces.spec.ts/space-panel-expanded-linux.png:1]: According to https://www.figma.com/design/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?node-id=764-5440&t=i9fX0TgxnYRwvN6j-0 we should switch to body lg semibold

File: playwright/snapshots/release-announcement/releaseAnnouncement.spec.ts/release-announcement-Threads-Activity-Centre-linux.png
Comment (robintown) [inline playwright/snapshots/release-announcement/releaseAnnouncement.spec.ts/release-announcement-Threads-Activity-Centre-linux.png:1]: Is this the desired heading appearance for toasts? It looks a little odd to my eyes

**Review by robintown — APPROVED** [review]
Non-blocking visual nit

File: playwright/snapshots/release-announcement/releaseAnnouncement.spec.ts/release-announcement-Threads-Activity-Centre-linux.png
Comment (florianduros) [reply playwright/snapshots/release-announcement/releaseAnnouncement.spec.ts/release-announcement-Threads-Activity-Centre-linux.png:1]: Yep too https://github.com/matrix-org/matrix-react-sdk/pull/12876#discussion_r1709890379

Comment (robintown) [reply playwright/snapshots/release-announcement/releaseAnnouncement.spec.ts/release-announcement-Threads-Activity-Centre-linux.png:1]: > we should switch to body lg semibold

Sounds good

**Comment by florianduros** [comment]
@robintown also the `mx_Heading_h4` heading is a lot bigger now. It doesn't looking really good in the right header playwright/snapshots/polls/polls.spec.ts/ThreadView-with-a-poll-on-group-layout-linux.png for example.

I wonder what is the best approach:
- Change the heading to body-lg-semibold
- Or only change in right header

**Comment by robintown** [comment]
I think the current pattern in which h1, h2, h3, h4 map directly to our heading/xl, heading/lg, heading/md, heading/sm design tokens is pretty nice. If we keep wanting to use body/\* tokens as headers, that's an indication that we should change something about our design system IMO. Germain noticed this too: https://github.com/element-hq/compound/issues/154

**Comment by florianduros** [comment]
> I think the current pattern in which h1, h2, h3, h4 map directly to our heading/xl, heading/lg, heading/md, heading/sm design tokens is pretty nice. If we keep wanting to use body/* tokens as headers, that's an indication that we should change something about our design system IMO. Germain noticed this too: [element-hq/compound#154](https://github.com/element-hq/compound/issues/154)

Hmmm you're right. I have the feeling that we are missing smaller tokens for heading. Maybe that's why designers are using body token for small headers.

Let's keep it in this state in this PR and we will have the conversation and update it if theses changes are not welcomed.

File: playwright/snapshots/user-onboarding/user-onboarding-new.spec.ts/User-Onboarding-new-user-page-is-shown-and-preference-exists-1-linux.png
Comment (t3chguy) [inline playwright/snapshots/user-onboarding/user-onboarding-new.spec.ts/User-Onboarding-new-user-page-is-shown-and-preference-exists-1-linux.png:1]: This looks too big

File: playwright/snapshots/user-view/user-view.spec.ts/user-info-linux.png
Comment (t3chguy) [inline playwright/snapshots/user-view/user-view.spec.ts/user-info-linux.png:1]: This looks far too big

**Review by t3chguy — APPROVED** [review]

**Pull request merged**